Push atmcorr_regr.py and atmcorr_specmath.py to cal in devel

helen-eifert opened the pull request:
Adds two more completed scripts in the pre processing pipeline. atmcorr_regr.py runs the regressions on .txt files of manually collected spectra from the radiance image and creates an output file that atmcorr_specmath.py uses to convert the radiance image to an atmospherically corrected radiance image. These scripts are used in between rad.py and refl.py. An experimental script for ROI detection was moved to a new branch and deleted from this branch because it is still in process. The ReadMe was also updated and Issue #29 was also addressed.

bspitzbart committed:
Add test notebook

helen-eifert committed:
Rename atmospheric_correction_update4.py to atmcorr_regr.py

Brian-Szutu committed:
Update atmcorr_regr.py

Brian-Szutu committed:
Update atmcorr_regr.py

Brian-Szutu committed:
Does spectral band math to radiance images

helen-eifert committed:
Rename spectral.py to atmcorr_specmath.py

helen-eifert committed:
Adds the correct file suffix

Since this will always be the last step of atmospheric correction, we can add "atmcorr" to the end of the file

Commit:
Merge pull request #30 from iceberg-project/devel

Sync atmcorr branch with latest devel

helen-eifert committed:
Fixes Issue #27

fixes specmather function and enables successful run of script!

helen-eifert committed:
Cleans up print statements in atmcorr_specmath.py

Brian-Szutu committed:
Fixes issue #31

Revises the check for the text containing average atmospheric correction values for each band.

Brian-Szutu committed:
Addresses issue #29

The script now looks up the date the image was taken in the XML file under the tag TLCTIME.
The variable "d" was changed to "dist" to make it more clear that it holds the Earth-Sun distance in AU.

helen-eifert committed:
Delete plot_cluster_iris.ipynb

bspitzbart commented:
Excellent! We will take a look and try to run it on bridges over the next couple of days.

helen-eifert committed:
Update README.md

helen-eifert committed:
Update README.md

helen-eifert committed:
Update README.md

bspitzbart commented:
Can you send an example .txt file? Is there an expected naming convention for the input txt files?

helen-eifert commented:
@bspitzbart I sent a sample file to you in Slack

bspitzbart committed:
add sample input .txt file

Brian-Szutu committed:
Fixes a potential syntax error in line 157

The .txt in line 157 should be .text to access a node's content

Brian-Szutu committed:
Fixes issue #34

Makes the output file name based on the information in the image xml file instead of the the folder the image is held in

Brian-Szutu committed:
Merge branch 'experiment/atmcorr' of https://github.com/iceberg-project/LandCover into experiment/atmcorr

iparask commented on src/data/orthoWV02_12FEB032148252-M1BS-1030010011973A00_u16ns3031_rad_atmcorr1.txt:
```diff
@@ -0,0 +1,160 @@
+ENVI ASCII Plot File [Thu Oct 25 12:42:53 2018]
```
May I ask the purpose of this file?

bspitzbart replied:
In the absence of automated region of interest (ROI) selection, this is the output from the manual process.  It is needed as input to atmcorr_regr.py.

iparask replied:
Would it make sense then to add it in a python file and import it? That way when this code becomes part of a package, we are sure that it gets installed correctly.

bspitzbart replied:
I don't think so.  It is a dynamic file.  The file will be produced by the ROI code and written to the path where atmcorr_regr.py expects to find it.  There will be a different file for each image, this is just an example. @helen-eifert Go ahead and push any others that you have for the other test images.

iparask replied:
Okay, I see!

helen-eifert replied:
@bspitzbart do you mean all the other manual output files? Would you like me to put them in the google drive or somewhere else?

bspitzbart commented:
Yes, I made a data folder in the repo.

On May 22, 2019 5:50 PM, "helen-eifert" <notifications@github.com> wrote:

*@helen-eifert* commented on this pull request.
------------------------------

In src/data/orthoWV02_12FEB032148252-M1BS-1030010011973A00_u16ns3031_
rad_atmcorr1.txt
<https://github.com/iceberg-project/LandCover/pull/32#discussion_r286702481>
:

> @@ -0,0 +1,160 @@
+ENVI ASCII Plot File [Thu Oct 25 12:42:53 2018]

@bspitzbart <https://github.com/bspitzbart> do you mean all the other
manual output files? Would you like me to put them in the google drive or
somewhere else?

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/iceberg-project/LandCover/pull/32?email_source=notifications&email_token=AAN5FWZ3OCJ45CPOY5G45U3PWW523A5CNFSM4HLDCBVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZN5FAI#discussion_r286702481>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/AAN5FW4LCRQ4IMIPV6XW5ZDPWW523ANCNFSM4HLDCBVA>
.

bspitzbart commented:
That is the ROI txt files, not images.

On May 22, 2019 5:56 PM, "Bradley Spitzbart" <
bradley.spitzbart@stonybrook.edu> wrote:

> Yes, I made a data folder in the repo.
>
> On May 22, 2019 5:50 PM, "helen-eifert" <notifications@github.com> wrote:
>
> *@helen-eifert* commented on this pull request.
> ------------------------------
>
> In src/data/orthoWV02_12FEB032148252-M1BS-1030010011973A00_
> u16ns3031_rad_atmcorr1.txt
> <https://github.com/iceberg-project/LandCover/pull/32#discussion_r286702481>
> :
>
> > @@ -0,0 +1,160 @@
> +ENVI ASCII Plot File [Thu Oct 25 12:42:53 2018]
>
> @bspitzbart <https://github.com/bspitzbart> do you mean all the other
> manual output files? Would you like me to put them in the google drive or
> somewhere else?
>
> —
> You are receiving this because you were mentioned.
>
> Reply to this email directly, view it on GitHub
> <https://github.com/iceberg-project/LandCover/pull/32?email_source=notifications&email_token=AAN5FWZ3OCJ45CPOY5G45U3PWW523A5CNFSM4HLDCBVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOBZN5FAI#discussion_r286702481>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAN5FW4LCRQ4IMIPV6XW5ZDPWW523ANCNFSM4HLDCBVA>
> .
>
>
>

helen-eifert committed:
Fixes Issue #35

Adds required packages to README

helen-eifert committed:
Adds test image text files for atmcorr_regr.py

Manually derived atmospheric correction spectra from ROIs used for regressions

helen-eifert commented:
@bspitzbart I've added all the sample txt files to the data folder (and also in the test image atmcorr folder in the google drive).

bspitzbart committed:
Update README.md

Add command line example

Brian-Szutu committed:
Added some additional comments in the header

The additional comments refer to the placeholder name of the output file of an image if there isn't an xml file to base its name off of. The placeholder name is "NO_XML_PRESENT"

Brian-Szutu committed:
Fixes issue #36

(Hopefully) fixes the issue where the script wouldn't detect the output .txt files of atmcorr_regr.py.
This script also relied on the name of the folder to look for the name of the output .txt.  This was changed so that it now looks for each image's associated xml file to find the the name of the output .txt.

iparask commented:
Since we merged Travis, do you think it makes sense to merge devel with this branch to see what Travis has to say?

bspitzbart commented:
Yes, we should.

bspitzbart committed:
Merge pull request #38 from iceberg-project/devel

Merge devel to incorporate Travis

bspitzbart commented:
Done. It gives a few good suggestions for things to clean up.

iparask commented:
Some should be fixed, other we will either change the rule or disable it. I will check it tomorrow and see what is too much and what not.

Commit:
Merge branch 'devel' into experiment/atmcorr

Commit:
Merge branch 'devel' into experiment/atmcorr

Commit:
Fixing flake8 and pylint

Commit:
Make some flake8 and pylint fixes

Commit:
more style edits

Brian-Szutu committed:
Fixes issue #41

Reverses the ordering on the importing of earth_sun_dist.py and adding the parent directory to the paths able to be imported from. Hopefully fixes the issue

helen-eifert committed:
Solves Issue #40

Adds gain & offset corrections from DigitalGlobe to radiance correction. Had trouble committing via Git Desktop for some reason.

Brian-Szutu committed:
Fixes issue #42

Changed the 0 to a 1 in sys.path.insert to hopefully fix the issue of refl.py not finding the earth_sun_dist dictionary

Brian-Szutu committed:
Merge branch 'experiment/atmcorr' of https://github.com/iceberg-project/LandCover into experiment/atmcorr

Brian-Szutu committed:
Addresses issue #43

The inputted directory from the console should now be a folder directly containing images and not other folders.
